moshi/src/main/java/com/squareup/moshi/ClassJsonAdapter.java

```diff
@@ -95,14 +95,15 @@ private void createFieldBindings(
         // Look up a type adapter for this type.
         Type fieldType = resolve(type, rawType, field.getGenericType());
         Set<? extends Annotation> annotations = Util.jsonAnnotations(field);
-        JsonAdapter<Object> adapter = moshi.adapter(fieldType, annotations);
+        String fieldName = field.getName();
+        JsonAdapter<Object> adapter = moshi.adapter(fieldType, annotations, fieldName);
```

Comment:
I did that deliberately. I assumed users would look for the Java field name instead of the JSON name.

I guess it depends on what the error message should look like? If somebody has a field `@Json("image_urls") ArrayList<HttpUrl> imageUrls`, "Error creating adapter for field User.Profile.imageUrls" seems better than "Error creating adapter for field User.Profile.image_urls"

Or should we treat it like a JSON path?

Reply:
i'll have to rethink the error message.
here's a case where the message would be confusing and wrong.
```
final class Foo {
  static final class Bar {
    ArrayList<String> field;
  }

  Map<String, Bar> map;
}
```
this would have some error message about "Foo.map.field"

Reply:
for posterity, the error message would now be about `field 'com.squareup.moshi.Moshi$Foo$Bar.field' in field 'com.squareup.moshi.Moshi$Foo.map'`.